symbolic-il2cpp/src/usym.rs

```diff
@@ -106,71 +200,86 @@ impl<'a> UsymSymbols<'a> {
     ///
     /// If `std::mem::size_of::<usize>()` is smaller than `std::mem::size_of::<u32>()` on
     /// the machine being run on.
-    pub fn parse(buf: &'a [u8]) -> Result<UsymSymbols<'a>> {
+    pub fn parse(buf: &'a [u8]) -> Result<UsymSymbols<'a>, UsymError> {
         if buf.as_ptr().align_offset(8) != 0 {
-            return Err(Error::msg("Data buffer not aligned to 8 bytes"));
+            return Err(UsymError::from(UsymErrorKind::MisalignedBuffer));
```

Comment:
You could make this even shorter with `return Err(UsymErrorKind::MisalignedBuffer.into())`, it's much muchness though.

Reply:
Good mulling.  I'd continue this general mulling along these lines:
- it's pretty obvious a `UsymErrorKind` probably belongs to a `UsymError`
- regardless, I don't really care what it coerces to when reading the code.  I know an error is being returned due to `return Err(...)` and I know that whatever it is, it is being identified as `MisalignedBuffer`.  This is plenty while reading even if I can't see the return type on screen.

Reply:
(I'm still not asking for any change btw, only discussing some thoughts around this)

Reply:
i think it makes sense to swap to using `into()` in that case: i've pushed a PR to do so. `map_or_else` at a glance didn't look like it was capable of inferring the type though, so i've kept the explicit `from` call in those places.